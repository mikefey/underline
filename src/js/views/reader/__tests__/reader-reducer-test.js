import test from 'ava';
import {
  LOAD_BOOK,
  LOAD_BOOK_COMPLETE,
  UPDATE_CURRENT_BOOK_DATA,
  READER_EXITED,
  BOOK_POSITION_DATA_UPDATED,
  SLIDER_RELEASED,
} from './../reader-constants';
import infoReducer from './../reader-reducer.js';
import bookFixture from './../../../../../test/fixtures/book';

test('default state', (t) => {
  t.snapshot(infoReducer(undefined, {}));
});

test('handles LOAD_BOOK', (t) => {
  t.snapshot(infoReducer(undefined, { type: LOAD_BOOK }));
});

test('handles LOAD_BOOK_COMPLETE', (t) => {
  t.snapshot(infoReducer(undefined, { type: LOAD_BOOK_COMPLETE }));
});

test('handles UPDATE_CURRENT_BOOK_DATA', (t) => {
  t.snapshot(infoReducer(undefined, { type: UPDATE_CURRENT_BOOK_DATA, data: bookFixture }));
});

test('handles READER_EXITED', (t) => {
  t.snapshot(infoReducer(undefined, { type: READER_EXITED }));
});

test('handles BOOK_POSITION_DATA_UPDATED', (t) => {
  t.snapshot(infoReducer(
    {
      currentBookData: bookFixture,
    },
    {
      type: BOOK_POSITION_DATA_UPDATED,
      data: { current_location: 0.96 },
    }));
});

test('handles SLIDER_RELEASED', (t) => {
  t.snapshot(infoReducer(undefined, { type: SLIDER_RELEASED, percentage: 0.57 }));
});
