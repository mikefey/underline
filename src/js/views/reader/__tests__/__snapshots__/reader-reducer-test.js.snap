// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default state 1`] = `
Object {
  "currentBookData": null,
  "currentBookId": null,
  "jumpToPage": null,
  "loadingBook": false,
  "sliderPercentageOnRelease": 0,
}
`;

exports[`handles BOOK_POSITION_DATA_UPDATED 1`] = `
Object {
  "currentBookData": Object {
    "author": "The book author",
    "cover_image": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
    "current_location": 0.96,
    "file": "blob",
    "id": 1,
    "parsed_data": null,
    "title": "A Good Book",
    "total_locations": 0,
  },
}
`;

exports[`handles LOAD_BOOK 1`] = `
Object {
  "currentBookData": null,
  "currentBookId": null,
  "jumpToPage": null,
  "loadingBook": true,
  "sliderPercentageOnRelease": 0,
}
`;

exports[`handles LOAD_BOOK_COMPLETE 1`] = `
Object {
  "currentBookData": null,
  "currentBookId": null,
  "jumpToPage": null,
  "loadingBook": false,
  "sliderPercentageOnRelease": 0,
}
`;

exports[`handles READER_EXITED 1`] = `
Object {
  "currentBookData": null,
  "currentBookId": null,
  "jumpToPage": null,
  "loadingBook": false,
  "sliderPercentageOnRelease": 0,
}
`;

exports[`handles SLIDER_RELEASED 1`] = `
Object {
  "currentBookData": null,
  "currentBookId": null,
  "jumpToPage": null,
  "loadingBook": false,
  "sliderPercentageOnRelease": 0.57,
}
`;

exports[`handles UPDATE_CURRENT_BOOK_DATA 1`] = `
Object {
  "currentBookData": Object {
    "author": "The book author",
    "cover_image": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
    "current_location": Object {
      "end": null,
      "percentage": 0,
      "start": null,
    },
    "file": "blob",
    "id": 1,
    "parsed_data": null,
    "title": "A Good Book",
    "total_locations": 0,
  },
  "currentBookId": null,
  "jumpToPage": null,
  "loadingBook": false,
  "sliderPercentageOnRelease": 0,
}
`;
